Doctrine/Phpcr/AutoRouteListener.php

```diff
@@ -73,6 +75,14 @@ public function onFlush(ManagerEventArgs $args)
                     $dm->persist($autoRoute);
                     $uow->computeChangeSets();
                 }
+
+                // reset locale to the original locale
+                if (null !== $locale) {
+                    $meta = $dm->getMetadataFactory()->getMetadataFor(get_class($document));
+                    $dm->findTranslation($meta->getName(), $meta->getIdentifierValue($document), $locale);
```

Comment:
you should use $uow->getDocumentId($document) instead. reading the $meta->getIdentifierValue code, it seems that the id could be not mapped on the document.